tests/unit/schema/test_api.py

```diff
@@ -2602,3 +2607,374 @@ def test_has_iam_mixin():
     })
     api_schema = api.API.build(fd, 'google.example.v1', opts=opts)
     assert api_schema.has_iam_mixin
+
+
+def get_file_descriptor_proto_for_method_settings_tests(
+    fields: Sequence[descriptor_pb2.FieldDescriptorProto] = None,
+    client_streaming: bool = False,
+    server_streaming: bool = False,
+) -> descriptor_pb2.FileDescriptorProto:
+    """
+    Args:
+        fields (Sequence[descriptor_pb2.FieldDescriptorProto]): Fields to include
+            in messages in the return object `descriptor_pb2.FileDescriptorProto`.
+        client_streaming (bool): Whether the methods in the return object
+            `descriptor_pb2.FileDescriptorProto` should use client streaming.
+        server_streaming (bool): Whether the methods in the return object
+            `descriptor_pb2.FileDescriptorProto` should use server streaming.
+    Return:
+        descriptor_pb2.FileDescriptorProto: Returns an object describing the API.
+    """
+    fd = (
+        make_file_pb2(
+            name="someexample.proto",
+            package="google.example.v1beta1",
+            messages=(
+                make_message_pb2(name="ExampleRequest", fields=fields),
+                make_message_pb2(name="ExampleResponse", fields=()),
+            ),
+            services=(
+                descriptor_pb2.ServiceDescriptorProto(
+                    name="SomeExample",
```

Comment:
nit, non-blocking: consider more descriptive names, like `ServiceOne` and `ServiceTwo` for theservices, and `DoFoo` for the methods. Just to make it easier for humans to track which names are used where. (`ExampleRequest` and `ExampleResponse` do that, which is great.)

Reply:
Fixed in https://github.com/googleapis/gapic-generator-python/pull/1975/commits/04fa73e4ab7801ddfdd0bdca34e47867b375a7fb